### PR TITLE
Add plugin for Fish shell tab completion

### PR DESCRIPTION
**Pull request by justinmayer**
As a long-time Fish shell aficionado, it is my pleasure to rebase the work that @jmwatte did on current master and submit his Fish shell tab completion plugin for Beets via this pull request.

I have tested the plugin's `beet fish` command and can confirm that the `~/.config/fish/completions/beet.fish` file that command generates does indeed result in the expected tab completion for `beet` commands in Fish shell.

Closes #1716

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,7 @@ Changelog
 
 New features:
 
+* A new :doc:`/plugins/fish` adds `Fish shell`_ tab autocompletion to Beets
```
Comment (sampsyo) [inline docs/changelog.rst]:
```suggestion
* A new :doc:`/plugins/fish` adds `Fish shell`_ tab autocompletion to beets.
```

File: docs/plugins/fish.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Usage
+-----
+
+Type ``beet fish`` to generate the ``beet.fish`` completions file at:
+``~/.config/fish/completions/``. If you later install or disable plugins, run
+``beet fish`` again to update the completions based on the enabled plugins.
+
+For users not accustomed to tab completion… After you type ``beet`` followed by
+a space in your shell prompt and then the ``TAB`` key, you should see a list of
+the Beets commands (and their abbreviated versions) that can be invoked in your
```
Comment (sampsyo) [inline docs/plugins/fish.rst]:
```suggestion
the beets commands (and their abbreviated versions) that can be invoked in your
```

File: docs/plugins/fish.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+current environment. Similarly, typing ``beet -<TAB>`` will show you all the
+option flags available to you, which also applies to subcommands such as
+``beet import -<TAB>``. If you type ``beet ls`` followed by a space and then the
+and the ``TAB`` key, you will see a list of all the album/track fields that can
+be used in Beets queries. For example, typing ``beet ls ge<TAB>`` will complete
```
Comment (sampsyo) [inline docs/plugins/fish.rst]:
```suggestion
be used in beets queries. For example, typing ``beet ls ge<TAB>`` will complete
```

File: docs/plugins/fish.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+to ``genre:`` and leave you ready to type the rest of your query.
+
+Options
+-------
+
+In addition to Beets commands, plugin commands, and option flags, the generated
```
Comment (sampsyo) [inline docs/plugins/fish.rst]:
```suggestion
In addition to beets commands, plugin commands, and option flags, the generated
```

File: docs/plugins/fish.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+completions also include by default all the album/track fields. If you only want
+the former and do not want the album/track fields included in the generated
+completions, use ``beet fish -f`` to only generate completions for Beets/plugin
```
Comment (sampsyo) [inline docs/plugins/fish.rst]:
```suggestion
completions, use ``beet fish -f`` to only generate completions for beets/plugin
```

**Comment by justinmayer** [comment]
Hey Adrian! Thanks for the kind words. I changed the capitalization where indicated and pushed the rebased commits. Speaking of commits, I tried to make them as discrete as possible and would recommend keeping them instead of squashing, but that's of course just my two cents. 😸 

This should be ready to merge now! 🚀

**Comment by sampsyo** [comment]
Wahoo! Thanks again! :sparkles: :rocket:

**Pull request merged**